.github/workflows/test.yaml

```diff
@@ -0,0 +1,20 @@
+on:
+  pull_request:
+  push:
+    branches: main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.stack
+            ./.stack-work
+          key: ${{ runner.os }}-${{ hashFiles('stack.yaml') }}-${{ hashFiles('*.cabal') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ hashFiles('stack.yaml') }}-
+            ${{ runner.os }}-
```

Comment:
This seems to be getting repeated. Should we have a `freckle/stack-cache-action`?

Reply:
I thought that from the very first project I typed it; I even started on one, but it's non trivial and I gave up. Feel free to check out the [npm library](https://github.com/actions/toolkit/tree/main/packages/cache), you would need to use to accomplish it.